kratos/includes/master_slave_constraint.h

```diff
@@ -0,0 +1,344 @@
+                                        const double Weight,
+                                        const double Constant) const
+    {
+        KRATOS_TRY
+
+        KRATOS_ERROR << "Create not implemented in MasterSlaveConstraintBaseClass" << std::endl;
+
+        KRATOS_CATCH("");
+    }
+
+    virtual MasterSlaveConstraint::Pointer Create(IndexType Id, NodeType& rMasterNode,
+                                        const VariableComponentType& rMasterVariable,
+                                        NodeType& rSlaveNode,
+                                        const VariableComponentType& rSlaveVariable,
+                                        const double Weight,
+                                        const double Constant) const
+    {
+        KRATOS_TRY
+
+        KRATOS_ERROR << "Create not implemented in MasterSlaveConstraintBaseClass" << std::endl;
+
+        KRATOS_CATCH("");
+    }
+
+
+    ///@}
+
+    ///@name Access
+    ///@{
+
+    /**
+	* Clears the maps contents
+	*/
+    void Clear()
+    {
+    }
+
+    /**
+     * is called to initialize the constraint
+     * if the constraint needs to perform any operation before any calculation is done
+     */
+    virtual void Initialize()
+    {
+    }
+
+    /**
+     * this is called in the beginning of each solution step
+     */
+    virtual void InitializeSolutionStep()
+    {
+    }
+
+
+    /**
+     * this is called for non-linear analysis at the beginning of the iteration process
+     */
+    virtual void InitializeNonLinearIteration()
+    {
+    }
+
+    /**
+     * this is called for non-linear analysis at the end of the iteration process
+     */
+    virtual void FinalizeNonLinearIteration()
+    {
+    }
+
+    /**
+     * this is called at the end of each solution step
+     */
+    virtual void FinalizeSolutionStep()
+    {
+    }
+
+
+    /**
+     * this determines the master equation IDs connected to this constraint
+     * @param rResult the elemental equation ID vector
+     * @param rCurrentProcessInfo the current process info instance
+     */
+    virtual void EquationIdVector(EquationIdVectorType& rSlaveEquationIds,
+                                  EquationIdVectorType& rMasterEquationIds,
+                                  ProcessInfo& rCurrentProcessInfo)
+    {
+        if (rSlaveEquationIds.size() != 0)
+            rSlaveEquationIds.resize(0);
+
+        if (rMasterEquationIds.size() != 0)
+            rMasterEquationIds.resize(0);
+    }
+
+    /**
+     * this is called during the assembling process in order
+     * to calculate all elemental contributions to the global system
+     * matrix and the right hand side
+     * @param rTransformationMatrix the elemental left hand side matrix
+     * @param rConstant the elemental right hand side
+     * @param rCurrentProcessInfo the current process info instance
+     */
+    virtual void CalculateLocalSystem(MatrixType& rTransformationMatrix,
```

Comment:
The transformation matrix is different from the local system right? Could we name this function differently?

Reply:
Thought the purpose of transformation matrix is different, it is also kind of local system. This is like  the provided Ke to the element into the constructor. 
I kept that name to be same as Element/Condition.